tests/python/pants_test/bin/test_runner_integration.py

```diff
@@ -37,3 +38,41 @@ def test_warning_filter(self):
     })
     self.assert_success(non_warning_run)
     self.assertNotIn('test warning', non_warning_run.stderr_data)
+
+  def test_parent_build_id_set_only_for_pants_runs_called_by_other_pants_runs(self):
+    with self.temporary_workdir() as workdir:
+      config = {
+        'GLOBAL': {
+          'colors': True,
+        }
+      }
+      command = [
+        'run',
+        'testprojects/src/python/nested_runs',
+        '--',
+        workdir,
+      ]
+      result = self.run_pants_with_workdir(
+        command,
+        workdir,
+        config,
+      )
+      self.assert_success(result)
+
+      run_tracker_dir = os.path.join(workdir, 'run-tracker')
+      self.assertTrue(os.path.isdir(run_tracker_dir), f'dir path {run_tracker_dir} does not exist!')
+      run_tracker_sub_dirs = (os.path.join(run_tracker_dir, dir_name) for dir_name in os.listdir(run_tracker_dir) if dir_name != 'latest')
+      for run_tracker_sub_dir in run_tracker_sub_dirs:
+        info_path = os.path.join(run_tracker_sub_dir, 'info')
+        self.assert_is_file(info_path)
+        with open(info_path, 'r') as info_f:
```

Comment:
Given this file is very consistently formatted in lines, I'd probably structure these checks at a slightly higher level, rather than seeking and doing a mix of per-line and per-whole-info checks. Maybe something like:
```python
with open(info_path, 'r') as info_f:
  lines = dict(line.split(": ", 1) for line in info_f.readlines())
  if "testprojects/src/python/nested_runs" in lines["cmd_line"]:
    self.assertIn("parent_build_id", lines)
  else:
    self.assertNotIn("parent_build_id", lines)
```